demisto_sdk/commands/download/downloader.py

```diff
@@ -95,647 +87,851 @@
+                    )
+                    return 1
+
+                all_custom_content_data = self.download_custom_content()
+                all_custom_content_objects = self.parse_custom_content_data(
+                    custom_content_data=all_custom_content_data
+                )
+
+                # Filter custom content so that we'll process only downloaded content
+                downloaded_content_objects = self.filter_custom_content(
+                    custom_content_objects=all_custom_content_objects
+                )
+
+                if not downloaded_content_objects:
+                    logger.info(
+                        "No custom content matching the provided input filters was found."
+                    )
+                    return 0
+
+                if self.auto_replace_uuids:
+                    # Replace UUID IDs with names in filtered content (only content we download)
+                    uuid_mapping = self.create_uuid_to_name_mapping(
+                        custom_content_objects=all_custom_content_objects
+                    )
+
+                    changed_uuids_count = 0
+                    for file_object in downloaded_content_objects.values():
+                        if self.replace_uuid_ids(
+                            custom_content_object=file_object, uuid_mapping=uuid_mapping
+                        ):
+                            changed_uuids_count += 1
+
+                    if changed_uuids_count > 0:
+                        logger.info(
+                            f"Replaced UUID IDs with names in {changed_uuids_count} custom content items."
+                        )
+
+            existing_pack_data = self.build_existing_pack_structure(
+                existing_pack_path=output_path
             )
-            is_valid = False
 
-        if self.system_item_type and not self.download_system_item:
-            logger.info(
-                "[red]The item type option is just for downloading system items.[/red]"
+            result = self.write_files_into_output_path(
+                downloaded_content_objects=downloaded_content_objects,
+                output_path=output_path,
+                existing_pack_structure=existing_pack_data,
             )
-            is_valid = False
 
-        return is_valid
+            if not result:
+                logger.error("Download failed.")
+                return 1
 
-    def handle_api_exception(self, e):
-        if e.status == 401:
-            logger.info(
-                "\n[red]Authentication error: please verify that the appropriate environment variables "
-                "(either DEMISTO_USERNAME and DEMISTO_PASSWORD, or just DEMISTO_API_KEY) are properly configured.[/red]\n"
-            )
-        logger.info(f"Exception raised when fetching custom content:\nStatus: {e}")
+            return 0
+
+        except Exception as e:
+            if not isinstance(e, HandledError):
+                logger.error(f"Error: {e}")
```

Comment:
A way to indicate that a specific error message has already been printed, and that we don't want to print an error message as it will be redundant.

Used for:
```python
except Exception as e:
    if isinstance(e, ApiException) and e.status == 401:
        logger.error(
            f"Server authentication error: {e}\n"
            "Please verify that the required environment variables ('DEMISTO_API_KEY', or "
            "'DEMISTO_USERNAME' and 'DEMISTO_PASSWORD') are properly configured."
        )

    elif isinstance(e, MaxRetryError):
        logger.error(
            f"Failed connecting to server: {e}.\n"
            "Please verify that the environment variable 'DEMISTO_BASE_URL' is properly configured, "
            "and that the server is accessible.\n"
            "If the server is using a self-signed certificate, try using the '--insecure' flag."
        )

    else:
        logger.error(f"Error while fetching custom content: {e}")

    raise HandledError from e
```
(The error messages are from previous code)